internal/k6runner/k6runner.go

```diff
@@ -92,15 +94,27 @@ var (
 func (r Processor) Run(ctx context.Context, registry *prometheus.Registry, logger logger.Logger, internalLogger zerolog.Logger) (bool, error) {
 	k6runner := r.runner.WithLogger(&internalLogger)
 
-	// TODO: This error message is okay to be Debug for local k6 execution, but should be Error for remote runners.
-	result, err := k6runner.Run(ctx, r.script)
+	k6Ctx, k6Span := trace.SpanFromContext(ctx).TracerProvider().Tracer("").Start(ctx, "k6 processor")
```

Comment:
This is also, again, the reason why I think you should store the tracer and not the tracerprovider in the context. This would become:

```
tracing.TracerFromContext(ctx).Start(ctx, "k6 processor")
```

Because the context contains a span, this creates a subspan with the current span as parent.

This aslo avoids the weird `Tracer("")` thing.

Reply:
I kid you not when I say I had exactly that code written, it was called `tracing.Tracer(ctx)` and it basically did `return tracing.TracerProvider(ctx).Tracer("")`.

I decied to remove it because as horrible as they may be, otel idioms and incantations are universal. The `""` thing is awful and the line is long, but it is the same thing everybody uses.
Very much like those nasty `panicIfNil(error)` functions you see in demo code. It is verbose, but it is the verbose everybody (that has used the otel sdk) will understand.

Reply:
> otel idioms and incantations are universal.

Please remind me to talk about this elsewhere.

Reply:
> Same argument as above: Storing the span in the context and retrieving the tracerprovider from it is "the otel way". I will do it like that everywhere I can.

I'll accept that, but then please remove the tracer provider from the context. It makes things hard to reason about.

Reply:
> I'll accept that, but then please remove the tracer provider from the context. It makes things hard to reason about.

Giving that a try, let's continue the discussion on the thread above.